### PR TITLE
fix: Make spacing between social buttons equal in all dimensions

### PR DESCRIPTION
**Pull request by Naman-1234**
Fixes ##6403

#### Short description of what this resolves:
Initially, Buttons was not equally spaced as well as they were of different sizes.

#### Changes proposed in this pull request:

-Made buttons equally spaced
-Made buttons of equal dimensions.

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

Here are some of the screenshots after change:

<h2>Screenshot in laptop size:</h2>

Screenshot (783) 

<h2>Screenshot of Surface Pro</h2>

Screenshot (784)

<h2>Screenshot of I Phone X</h2>

Screenshot (785)
<br><br><br></br></br></br>

### CODE REVIEW

**Review by himanshujaidka — COMMENTED** [review]
I think in mobile view the space is too much, adjust it according to size of screen

**Comment by Naman-1234** [comment]
Ok Working on it

**Comment by Naman-1234** [comment]
@himanshujaidka Sir, should I make the buttons bigger according to screen or decrease space?

**Comment by himanshujaidka** [comment]
> @himanshujaidka Sir, should I make the buttons bigger according to screen or decrease space?

I am also a participant and learner like you so don't call me sir and answer to your question is that I think it should be matched according to the screen-size

**Comment by himanshujaidka** [comment]
> Fixes ##6403
> 
> #### Short description of what this resolves:
> Initially, Buttons was not equally spaced as well as they were of different sizes.
> 
> #### Changes proposed in this pull request:
> -Made buttons equally spaced
> -Made buttons of equal dimensions.
> 
> #### Checklist
> * [x]  I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
> * [ ]  My branch is up-to-date with the Upstream `development` branch.
> * [ ]  The acceptance, integration, unit tests and linter pass locally with my changes
> * [ ]  I have added tests that prove my fix is effective or that my feature works
> * [ ]  I have added necessary documentation (if appropriate)
> 
> Here are some of the screenshots after change:
> 
> ## Screenshot in laptop size:
> Screenshot (783)
> 
> ## Screenshot of Surface Pro
> Screenshot (784)
> 
> ## Screenshot of I Phone X
> Screenshot (785)

in this button is suited to the screen but in mobile view, it is bigger than the other materials so in small screens the size of buttons should be small

**Comment by Naman-1234** [comment]
Ok Working on it

**Review by vsasvipul0605 — DISMISSED** [review]

**Review by MeghalBisht — COMMENTED** [review]
Now the icons are not centered inside the button and there are unnecessary gaps created b/w 2 buttons. 
cc

**Comment by mariobehling** [comment]
Basically, the goal is to avoid the irregular distribution of buttons. The goal is - not - to change the buttons itself or to justify buttons. The alignment should neither be changed.

File: app/styles/libs/_helpers.scss [hunk 1]
```diff
@@ -130,3 +130,10 @@ $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
 .text-white {
   color: #fff;
 }
+
+.mybtn
+{
```
Comment (nlok5923) [inline app/styles/libs/_helpers.scss:135]: please stick to the formatting style .

**Review by nlok5923 — CHANGES_REQUESTED** [review]

File: app/styles/libs/_helpers.scss (hunk 1 above)
Comment (Naman-1234) [reply app/styles/libs/_helpers.scss:135]: Changes done

**Comment by Naman-1234** [comment]
Changes requested done
Ss after the changes are:
Screenshot (852)<br></br><br></br><br></br>
Screenshot (850)

**Comment by Naman-1234** [comment]
@mariobehling Please review this Sir, I have done all the changes told

**Comment by Naman-1234** [comment]
@iamareebjamal  and @mariobehling  Sir, Please review this PR. I have made all the changes told in the issue.

**Comment by mariobehling** [comment]
Thanks, looks already much better. Please also ensure the spacing between widget areas is consistent.

Spacing between link box area, event filters, and tweets is larger than the spacing between the top areas. Please adjust this as well.

Screenshot from 2021-01-21 13-42-27

**Comment by Naman-1234** [comment]
Sure sir

**Comment by Naman-1234** [comment]
@mariobehling  Sir,I think all looks good now, As the spacing between widget is constant, especially that between "Add to Calender" button and its above is same as that list items in Tracks list. 

Screenshot (864)

**Comment by mariobehling** [comment]
You increased the spacing everywhere, but you should have made the same size it was already in the top areas before. Do not increase it.

Screenshot from 2021-01-21 23-43-03

**Comment by Naman-1234** [comment]
@mariobehling  Sir Please review this Pr, Changes as per told by you have been done, Now:
1. Spacing between different sections is consistent.
2. Spacing between the upper section is not increased as told by you.
Here is the screenshot of the current look:



Screenshot (877)

**Comment by mariobehling** [comment]
Thanks. As this area has been changed elsewhere, please resolve merge conflicts.

**Comment by Naman-1234** [comment]
@mariobehling  Sir all the merge conflicts have been removed.

**Comment by mariobehling** [comment]
Conflicts have been removed, but spacing is too big and unequal. You were closer to a solution in earlier PRs it seems.

Screenshot from 2021-01-26 09-31-21

**Comment by Naman-1234** [comment]
@mariobehling  Sir ,Now I have updated my branch to the latest so there are no merge conflicts and the changes have been again done which were approved by you before merge conflicts. Please take a look here
Screenshot (882)

File: app/styles/libs/_helpers.scss [hunk 2]
```diff
@@ -104,10 +104,6 @@ $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
   white-space: nowrap;
 }
 
-.pre-line {
```
Comment (iamareebjamal) [inline app/styles/libs/_helpers.scss:107]: Why?

Comment (Naman-1234) [reply app/styles/libs/_helpers.scss:107]: Sir, the Reason is it was giving extra space as you can see in the screenshot given by Mario Sir, And sir told me to remove those extra spaces and make it look like the screenshot given earlier by me.

Comment (iamareebjamal) [reply app/styles/libs/_helpers.scss:107]: So you'll remove the class? Imagine if doctors remove the heart of the patient if they have heart cancer

Comment (Naman-1234) [reply app/styles/libs/_helpers.scss:107]: So sorry sir, I will undo the changes and will try to do it without deleting the classes. Once again I am so sorry for such thing.

**Comment by Naman-1234** [comment]
@mariobehling  Sir, Changes have been done. Please review this PR. A screenshot of the final looking is given below. Changes have been taken to the last time you accepted changes and told me to remove merge conflicts. This time I have taken precautions and have not removed any classes.

Screenshot (891)

File: app/styles/libs/_helpers.scss (hunk 2 above)
Comment (iamareebjamal) [reply app/styles/libs/_helpers.scss:107]: The class is still removed

File: app/styles/libs/_helpers.scss [hunk 3]
```diff
@@ -131,10 +127,24 @@ $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
...
+  height: 50px;
+  max-width: 50px;
+  max-height: 50px;
+}
+
 .text-white {
   color: #fff;
 }
 
-.w-full {
```
Comment (iamareebjamal) [inline app/styles/libs/_helpers.scss:138]: Why

Comment (Naman-1234) [reply app/styles/libs/_helpers.scss:138]: Oh sorry sir, Actually I forgot to add these changes in the commit, I removed them for testing purpose but forgot to add them again after it,  I will just add and then do the Pr, Thank you so much for reminding me sir.

Comment (Naman-1234) [reply app/styles/libs/_helpers.scss:138]: Sir I have added them again and the changes are still there , As you can see in the screenshot below:

Screenshot (892)

File: app/templates/public.hbs [hunk 4]
```diff
@@ -67,7 +67,7 @@
               </div>
           {{/if}}
           {{#if (eq this.session.currentRouteName 'public.sessions')}}
-            <div class="{{if this.device.isMobile 'd-flex'}} ui basic segment">
+            <div class="{{if this.device.isMobile 'd-flex'}} ui basic segment" style="margin-top: 0px;   padding-top:0px">
```
Comment (iamareebjamal) [inline app/templates/public.hbs:70]: Use class

File: app/styles/libs/_helpers.scss [hunk 5]
```diff
@@ -131,10 +131,28 @@ $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
   width: fit-content;
 }
 
-.text-white {
```
Comment (iamareebjamal) [inline app/styles/libs/_helpers.scss:134]: Why

File: app/templates/public.hbs (hunk 4 above)
Comment (Naman-1234) [reply app/templates/public.hbs:70]: Ok sir

File: app/styles/libs/_helpers.scss (hunk 5 above)
Comment (Naman-1234) [reply app/styles/libs/_helpers.scss:134]: Will add in the latest PR sir, Sorry for inconvenience

Comment (Naman-1234) [reply app/styles/libs/_helpers.scss:134]: Sir due to addition of css in the middle this styling is shifted to line 145

Comment (Naman-1234) [reply app/styles/libs/_helpers.scss:134]: Sir I did the changes , Please review it again.

Comment (iamareebjamal) [reply app/styles/libs/_helpers.scss:134]: This class is still removed

Comment (Naman-1234) [reply app/styles/libs/_helpers.scss:134]: Sir, I am not able to understand how this class is removed,  as you can see in the screenshot given,  It just shifed to line 145 from 133 ,not removed.
Screenshot (893)

Comment (iamareebjamal) [reply app/styles/libs/_helpers.scss:134]: Revert it to where it was

Comment (Naman-1234) [reply app/styles/libs/_helpers.scss]: Done sir

Comment (Naman-1234) [reply app/styles/libs/_helpers.scss]: @iamareebjamal  and @mariobehling  Sir Please review this PR.

File: app/styles/libs/_helpers.scss [hunk 6]
```diff
@@ -135,6 +135,24 @@ $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
...
+  height: 50px;
+  max-width: 50px;
+  max-height: 50px;
+}
+
 .w-full {
   width: 100% !important;
 }
+
+.social-link-button {
```
Comment (iamareebjamal) [inline app/styles/libs/_helpers.scss:149]: Duplicate style. Next time, review the PR yourself once before asking for review

File: app/styles/libs/_helpers.scss [hunk 7]
```diff
@@ -135,6 +135,18 @@ $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
...
+}
+
 .w-full {
   width: 100% !important;
 }
+
+.custom-padding {
+  margin-top: 0 !important;
+  padding-top: 0 !important;
+}
```
Comment (iamareebjamal) [inline app/styles/libs/_helpers.scss:152]:
```suggestion
```

File: app/templates/components/public/social-links.hbs [hunk 8]
```diff
@@ -1,31 +1,31 @@
...
+          <a href="{{socialLink.link}}" class="ui icon blue mb-1 social-link-button button extra-link d-flex items-center content-center" target="_blank" rel="noopener nofollow">
             <div  class="clamp" style="font-size: .8rem;">{{socialLink.displayName}}</div>
           </a>
         </UiPopup>
       </div>
     {{/if}}
   {{/each}}
   </div>
-</div>
+</div>
```
Comment (iamareebjamal) [inline app/templates/components/public/social-links.hbs:31]:
```suggestion
</div>

```

**Review by mariobehling — APPROVED** [review]
Thank you! Glad this issue is finished.

**Pull request merged**